src/interactive.js

```diff
@@ -1,75 +1,187 @@
+    const list = document.querySelector('#todo-data');
+    if (todoList !== null) {
       todoList.forEach((task) => {
         const li = document.createElement('li');
         if (task.completed === true) {
           const text = `<div class="list-container"> <input class='check-input' type='checkbox' value='${task.completed}' checked>
-                <p class="todo-text">${task.itemText}</p><a class="del-menu" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+                <p class="todo-text" contenteditable="true">${task.todoText}</p><div class="del-menu"><a class="edit" href="#">
+                <i class="fas fa-edit edit-box"></i></a><a href="#" class="del delete"><i class="fas fa-trash del-box"></i></a></div></div>`;
           li.classList.add('list-item');
           li.innerHTML = text;
           list.appendChild(li);
         } else {
           const text = `<div class="list-container"> <input class='check-input' type='checkbox' value='${task.completed}'>
-              <p class="todo-text">${task.itemText}</p><a class="del-menu" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+              <p class="todo-text" contenteditable="true">${task.todoText}</p><div class="del-menu"><a class="edit" href="#">
+              <i class="fas fa-edit edit-box"></i></a><a href="#" class="del delete"><i class="fas fa-trash del-box"></i></a></div></div>`;
           li.classList.add('list-item');
           li.innerHTML = text;
           list.appendChild(li);
         }
       });
+      const check = document.querySelectorAll('.check-input');
+      const todos = JSON.parse(localStorage.getItem('todos'));
+      check.forEach((check, index) => {
+        check.addEventListener('change', (e) => {
+          e.preventDefault();
+          const divWrapper = check.parentElement;
+          if (check.checked) {
+            divWrapper.classList.add('checked');
+            check.checked = true;
+            todos[index].completed = true;
+            this.addItem(todos);
+          } else {
+            divWrapper.classList.remove('checked');
+            todos[index].completed = false;
+            this.addItem(todos);
+          }
+        });
+        window.addEventListener('load', (e) => {
+          e.preventDefault();
+          const divWrapper = check.parentElement;
+          if (check.checked) {
+            divWrapper.classList.add('checked');
+            check.checked = true;
+            todos[index].completed = true;
+            this.addItem(todos);
+          } else {
+            divWrapper.classList.remove('checked');
+            todos[index].completed = false;
+            this.addItem(todos);
+          }
+        });
+      });
     }
+  }
 
-    const check = document.querySelectorAll('.check-input');
-    const todos = JSON.parse(localStorage.getItem('todos'));
-    check.forEach((check, index) => {
-      check.addEventListener('change', (e) => {
+  editItem() {
```

Comment:
[ OPTIONAL ] This function does not work as the user is unable to edit the description of the item. Please fix the problem.

Reply:
It works if you click the edit button after editing the task description.